chapters/overfitting.qmd

```diff
@@ -0,0 +1,487 @@
+
+pointers_blowback <- 
+  tibble(A = c(2.20),  B = c(-3.5),
+         C = c(1.25),  D = c(-3.5))
+
+high_plot <- 
+  base_plot  +
+  geom_contour(
+    data = high_cost,
+    aes(z = value),
+    breaks = 0,
+    col = "black",
+    linewidth = 1 / 2,
+    alpha = 3 / 4
+  )  +
+  geom_segment(data = pointers,
+               aes(xend = C, yend = D),
+               arrow  = arrow(length = unit(0.03, "npc"))) +
+  geom_segment(data = pointers_blowback,
+               aes(xend = C, yend = D),
+               color = "red",
+               arrow  = arrow(length = unit(0.03, "npc"))) +
+  ggtitle("(d) High complexity")  +
+  labs(x = "Predictor A", y = NULL)
+
+mid_plot <- 
+  base_plot  +
+  geom_contour(
+    data = mid_cost,
+    aes(z = value),
+    breaks = 0,
+    col = "black",
+    linewidth = 1 / 2,
+    alpha = 3 / 4
+  )  +
+  ggtitle("(c) Medium complexity") +
+  labs(x = "Predictor A", y = NULL)
+
+low_plot <- 
+  base_plot  +
+  geom_contour(
+    data = low_cost,
+    aes(z = value),
+    breaks = 0,
+    col = "black",
+    linewidth = 1 / 2,
+    alpha = 3 / 4
+  )  +
+  ggtitle("(b) Low complexity") +
+  labs(x = "Predictor A", y = NULL)
+
+base_plot + low_plot + mid_plot + high_plot + plot_layout(nrow = 1, byrow = FALSE)
+```
+
+When the cost is increased, @fig-two-class-overfit(c) shows an increase in complexity. The boundary is more parabolic, and a few more training set points are on the correct side of the estimated line. 
+
+@fig-two-class-overfit(d) shows what occurs when the model is instructed that it is incredibly bad when incorrectly predicting a data point in the training set. Its boundary contorts in a manner that attempts to gain a few more points of accuracy. Two specific points, identified by black arrows, appear to drive much of the additional structure in the black curve. Since we know the actual boundary, it's easy to see that these contortions will not reproduce with new data. For example, it is implausible that a small island of vermilion circles exists in the mainstream of blue triangles. 
+
+Another sign that a model probably has too much complexity is the additional component of the decision boundary at the bottom of @fig-two-class-overfit(d) (red arrow). Points inside this part of the boundary would be classified as triangles even though no triangles are near it. This is a sort of "blowback" of an overly complex model where choices the model makes in one part of the predictor space add complexity in a completely different part of the space. This blowback is often seen in areas where there are no data. It’s unnecessary and points to the idea that the model is trying to do too much. 
+
+This boundary demonstrates how added complexity can be a bad thing. A test set of data was simulated and is shown in @fig-two-class-new. The performance for this overly complex model is poor since the extra areas induced by the high cost value do not contain the right class of points. 
```

Comment:
```suggestion
This boundary demonstrates how added complexity can be a bad thing. A test set of data was simulated and is shown in @fig-two-class-new. The performance for this overly complex model is far from optimal since the extra areas induced by the high cost value do not contain the right class of points. 
```